vrs_anvil/evidence.py

```diff
@@ -0,0 +1,370 @@
+import io
+import json
+import os
+from pathlib import Path
+import sqlite3
+import pandas as pd
+
+from datetime import datetime
+from firecloud import api as fapi
+from pysam import VariantFile, VariantRecord
+
+
+def get_vcf_row(
+    variant_id: str, vcf: VariantFile, index_path: str = None
+) -> VariantRecord:
+    """given a variant id and annotated VCF, get the row ID associated with it
+
+    Args:
+        variant_id (str): VRS ID for the variant of interest
+        vcf (VariantFile): Pysam VariantFile
+        index_path (str, optional): Index used to speed up search for variant. Defaults to iterating through VCF.
+
+    Raises:
+        Exception: outputs if no index is found
+
+    Returns:
+        VariantRecord: A Pysam VariantRecord (VCF row)
+    """
+
+    assert "VRS_Allele_IDs" in vcf.header.info, (
+        "no VRS_Allele_IDs key in INFO found, "
+        "please ensure that this is an VRS annotated VCF"
+    )
+
+    # try to populate from Bash env variable
+    if not index_path:
+        assert (
+            "VRS_VCF_INDEX" in os.environ
+        ), "no genotype index specified, no index path was provided nor was a variable name VRS_VCF_INDEX found."
+        index_path = Path(os.environ.get("VRS_VCF_INDEX"))
+
+    if index_path:
+        # if index provided, use it to get VCF row
+        index_path = Path(index_path)
+
+        # find variant of interest
+        for _, chr, pos in fetch_by_vrs_ids([variant_id], index_path):
+            for record in vcf.fetch(f"chr{chr}", pos - 1, pos):
```

Comment:
do VCF `chr` fields have to follow the UCSC `chr<Z>` naming convention or can they also just be `<Z>` (ie `"21"` instead of `"chr21"`)? if so, might be something to flag in a README or catch early on?

Reply:
I have had a related lingering concern in my brain about the relationship between a chromosome identifier and a file in the context of using the VRS index for a bit. Made an issue just to put it on paper: https://github.com/gks-anvil/vrsix/issues/2

Reply:
Good point. Will add the prerequisites of the VCF in the readme.

Reply:
see [README](https://github.com/ohsu-comp-bio/vrs_anvil_toolkit/blob/7dcf7d1f1dd35378bfb7b133dbe0e3f257372197/README.md/#L111-L113)